### PR TITLE
HW7

### PR DESCRIPTION
**Pull request by Amudah41**

### CODE REVIEW

File: hw7/tasks/task72.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    Input: a = "a#c", t = "b"
+    Output: False
+    Explanation: s becomes "c" while t becomes "b".
+
+"""
+from itertools import zip_longest
+
+
+def yeild_str(my_str: str):
```
Comment (AlekseiAfanasev) [inline hw7/tasks/task72.py]: name does not show what this function does

File: hw7/tasks/task72.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    my_str = iter(reversed(my_str))
+    for a in my_str:
+        if a == "#":
+            try:
+                _ = next(my_str)
```
Comment (AlekseiAfanasev) [inline hw7/tasks/task72.py]: "_" is used only for meaningless variables, try to rename

File: hw7/tasks/task72.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                while _ == "#":
+                    _ = next(my_str)
+            except StopIteration:
+                break
+            continue
+        print(a, my_str)
```
Comment (AlekseiAfanasev) [inline hw7/tasks/task72.py]: probably forgot to remove

File: hw7/tests/task72_test.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        ("###", []),
+    ],
+)
+def test_yeild_str(input_str: str, expected_result: List[str]):
+    actual_result = [item for item in yeild_str(input_str)]
+    assert actual_result == expected_result
+
+
+def test_positive_backspace_compare_with_example_value_1():
+    assert backspace_compare(s="ab#c", t="ad#c") == True
```
Comment (AlekseiAfanasev) [inline hw7/tests/task72_test.py]: it's enough to check "assert x" and "assert not x".
"== True" and "== False" are the same

Comment (Amudah41) [reply hw7/tests/task72_test.py]: Did it in that way, because Andrey said, that tests should be as clearer as possible

**Review by AlekseiAfanasev — APPROVED** [review]

**Pull request merged**